amd_openvx_extensions/amd_rpp/source/ResizeMirrorNormalizeTensor.cpp

```diff
@@ -0,0 +1,343 @@
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#include "internal_publishKernels.h"
+
+struct ResizeMirrorNormalizeTensorLocalData {
+	RPPCommonHandle handle;
+	rppHandle_t rppHandle;
+	Rpp32u device_type;
+	Rpp32u nbatchSize;
+	RppiSize *srcDimensions;
+	RppiSize maxSrcDimensions;
+	RppiSize *dstDimensions;
+	RppiSize maxDstDimensions;
+	RppPtr_t pSrc;
+	RppPtr_t pDst;
+	vx_float32  *mean;
+	vx_float32 *std_dev;
+	vx_uint32 *mirror;
+	vx_uint32 chnShift; //NHWC to NCHW
+	Rpp32u *srcBatch_width;
+	Rpp32u *srcBatch_height;
+	Rpp32u *dstBatch_width;
+	Rpp32u *dstBatch_height;
+	RpptDescPtr srcDescPtr, dstDescPtr;
+    RpptROIPtr roiTensorPtrSrc;
+    RpptRoiType roiType;
+    RpptImagePatchPtr dstImgSize;
+    RpptDesc srcDesc, dstDesc;
+#if ENABLE_HIP
+	void *hip_pSrc;
+	void *hip_pDst;
+	RpptImagePatch *d_dstImgSize;
+	RpptROI *d_roiTensorPtrSrc;
+#endif
+};
+
+static vx_status VX_CALLBACK refreshResizeMirrorNormalizeTensor(vx_node node, const vx_reference *parameters, vx_uint32 num, ResizeMirrorNormalizeTensorLocalData *data)
+{
+	vx_status status = VX_SUCCESS;
+	STATUS_ERROR_CHECK(vxCopyArrayRange((vx_array)parameters[6], 0, data->nbatchSize * 3, sizeof(vx_float32), data->mean, VX_READ_ONLY, VX_MEMORY_TYPE_HOST));
+	STATUS_ERROR_CHECK(vxCopyArrayRange((vx_array)parameters[7], 0, data->nbatchSize * 3, sizeof(vx_float32), data->std_dev, VX_READ_ONLY, VX_MEMORY_TYPE_HOST));
+	STATUS_ERROR_CHECK(vxCopyArrayRange((vx_array)parameters[8], 0, data->nbatchSize, sizeof(vx_uint32), data->mirror, VX_READ_ONLY, VX_MEMORY_TYPE_HOST));
+    STATUS_ERROR_CHECK(vxReadScalarValue((vx_scalar)parameters[9], &data->chnShift));
+	STATUS_ERROR_CHECK(vxCopyArrayRange((vx_array)parameters[1], 0, data->nbatchSize, sizeof(Rpp32u), data->srcBatch_width, VX_READ_ONLY, VX_MEMORY_TYPE_HOST));
+	STATUS_ERROR_CHECK(vxCopyArrayRange((vx_array)parameters[2], 0, data->nbatchSize, sizeof(Rpp32u), data->srcBatch_height, VX_READ_ONLY, VX_MEMORY_TYPE_HOST));
+	STATUS_ERROR_CHECK(vxCopyArrayRange((vx_array)parameters[4], 0, data->nbatchSize, sizeof(Rpp32u), data->dstBatch_width, VX_READ_ONLY, VX_MEMORY_TYPE_HOST));
+	STATUS_ERROR_CHECK(vxCopyArrayRange((vx_array)parameters[5], 0, data->nbatchSize, sizeof(Rpp32u), data->dstBatch_height, VX_READ_ONLY, VX_MEMORY_TYPE_HOST));
+	for(int i = 0; i < data->nbatchSize; i++) {
+		data->srcDimensions[i].width = data->roiTensorPtrSrc[i].xywhROI.roiWidth = data->srcBatch_width[i];
```

Comment:
I don't think data->srcDimensions is used. Check if all these variables are used. If not remove them.

Reply:
Removed the unused variables from the file